Mention remoteEnv workaround in docs for error 'The configured global bin directory .../.local/share/pnpm/bin is not in PATH'

I tried to install pnpm first using `ghcr.io/devcontainers-extra/features/pnpm:2` and I assume this would be the main alternative when not using the Node base image. But this didn't work by default and I filed an issue for the pnpm issue here: https://github.com/devcontainers-extra/features/issues/218

I added this to my devcontainer file as a workaround in the meantime. Perhaps this workaround could be added to the docs for `mount-pnpm-store`?

```json
	// Workaround: the pnpm feature adds pnpm to PATH only for interactive
	// terminals (via ~/.bashrc etc.), so devcontainer lifecycle hooks like
	// onCreateCommand can't find it. Set PATH here so mount-pnpm-store's
	// onCreate script can locate pnpm's global bin dir.
	"remoteEnv": {
		"PNPM_HOME": "/home/vscode/.local/share/pnpm",
		"PATH": "/home/vscode/.local/share/pnpm/bin:${containerEnv:PATH}"
	}
```

Otherwise I get this error:

```
Setting pnpm store location to /.pnpm-store
[ERROR] The configured global bin directory "/home/vscode/.local/share/pnpm/bin" is not in PATH
Run "pnpm setup" to update your shell configuration.


[2026-05-08T00:20:52.666Z] mount-pnpm-store-setup of onCreateCommand from Feature 'ghcr.io/joshuanianji/devcontainer-features/mount-pnpm-store:1' failed with exit code 1. Skipping any further user-provided commands.
[2026-05-08T00:20:52.667Z] Error: Command failed: /bin/sh -c /usr/local/share/mount-pnpm-store/scripts/oncreate.sh
```

--- a/src/mount-pnpm-store/NOTES.md
+++ b/src/mount-pnpm-store/NOTES.md
@@ -38,9 +38,10 @@ The volume mount is called `global-devcontainer-pnpm-store`, so ensure that no o
 
 ## Changelog
 
-| Version | Notes                                           |
-| ------- | ----------------------------------------------- |
-| 1.0.2   | Move onCreate lifecycle script to `oncreate.sh` |
+| Version | Notes                                                      |
+| ------- | ---------------------------------------------------------- |
+| 1.0.3   | Fix pnpm global bin dir check in `oncreate.sh` (issue #80) |
+| 1.0.2   | Move onCreate lifecycle script to `oncreate.sh`            |
 | 1.0.1   | Fix Docs                                        |
 | 1.0.0   | Support zsh + refactor                          |
 | 0.1.1   | Fix mount name                                  |

--- a/src/mount-pnpm-store/devcontainer-feature.json
+++ b/src/mount-pnpm-store/devcontainer-feature.json
@@ -1,7 +1,7 @@
 {
     "name": "Mount pnpm Store",
     "id": "mount-pnpm-store",
-    "version": "1.0.2",
+    "version": "1.0.3",
     "documentationURL": "https://github.com/joshuanianji/devcontainer-features/tree/main/src/mount-pnpm-store",
     "description": "Sets pnpm store to ~/.pnpm-store and mounts it to a volume to share between multiple devcontainers",
     "options": {},

--- a/src/mount-pnpm-store/oncreate.sh
+++ b/src/mount-pnpm-store/oncreate.sh
@@ -2,9 +2,18 @@
 
 set -e
 
+# pnpm refuses --global commands (exit 1) when its global bin directory is
+# not in PATH. This script runs as onCreateCommand in a non-interactive shell
+# where rc-file exports are never loaded, so provide the environment here.
+# The expected directory differs by pnpm version ($PNPM_HOME for pnpm <= 8,
+# $PNPM_HOME/bin for pnpm >= 9), so put both on PATH; neither needs to exist
+# for the check to pass.
+export PNPM_HOME="${PNPM_HOME:-$HOME/.local/share/pnpm}"
+export PATH="$PNPM_HOME/bin:$PNPM_HOME:$PATH"
+
 # set pnpm config (if it's installed)
 if type pnpm >/dev/null 2>&1; then
-    echo "Setting pnpm store location to $_REMOTE_USER_HOME/.pnpm-store"
+    echo "Setting pnpm store location to $HOME/.pnpm-store"
     pnpm config set store-dir ~/.pnpm-store --global
 else
     echo "WARN: pnpm is not installed! Please ensure pnpm is installed and in your PATH."

--- a/test/mount-pnpm-store/node_v2.sh
+++ b/test/mount-pnpm-store/node_v2.sh
@@ -1,0 +1,16 @@
+#!/bin/bash
+
+set -e
+
+# Regression test for the pnpm >= 9 global bin dir check (issue #80):
+# node feature v2 installs latest pnpm by default, which made the
+# unpatched oncreate.sh fail the container build with
+# "The configured global bin directory ... is not in PATH".
+# Reaching this script at all proves onCreateCommand survived.
+
+# The test shell is non-interactive too, so it needs the same env
+# the patched oncreate.sh provides for itself.
+export PNPM_HOME="${PNPM_HOME:-$HOME/.local/share/pnpm}"
+export PATH="$PNPM_HOME/bin:$PATH"
+
+./_default.sh

--- a/test/mount-pnpm-store/scenarios.json
+++ b/test/mount-pnpm-store/scenarios.json
@@ -12,6 +12,13 @@
             "mount-pnpm-store": {}
         }
     },
+    "node_v2": {
+        "image": "mcr.microsoft.com/devcontainers/base:debian-12",
+        "features": {
+            "ghcr.io/devcontainers/features/node:2": {},
+            "mount-pnpm-store": {}
+        }
+    },
     "zsh_shell": {
         "image": "mcr.microsoft.com/devcontainers/javascript-node:1-18",
         "features": {

